xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocumentsJSON.xml

```diff
@@ -60,11 +60,15 @@ $response.setContentType('application/json')
   #set ($limit = 15)
 #end
 #set($docNameFilter = $request.get('ddoc.fullName'))
+#set($dateFilter = $request.get('ddoc.date'))
 #set($deleterFilter = $request.get('ddoc.deleter'))
 #set($sql = '')
 #if("$!{docNameFilter}" != '')
   #set($sql = "and lower(ddoc.fullName) like '%${docNameFilter.trim().toLowerCase()}%' ")
 #end
+#if("$!{dateFilter}" != '')
+  #set($sql = $sql + "and ddoc.date between '%${dateFilter.split("-")[0]}%' and '%${dateFilter.split("-")[1]}%' ")
```

Comment:
Same as above, it would be nice to use query parameters. Moreover, you're assuming that the date filter is always a range, but it could be also some free text (see for instance [XWIKI-14909](https://jira.xwiki.org/browse/XWIKI-14909)). Here's an [example of code](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml#L706) that handles both cases (range and free text).